### DISCUSSION TITLE
Licence that disallows contributing

### DISCUSSION THREAD

**Discussion by PabloPomodoro** [community/community#118729] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I would like to add a licence to my repository, that either disallows contributing or at least doesn't permit it, so that I can decline potential PRs.

Would adding a MIT licence + stating, that this repo is my personal project therefore I don't allow contributions, be the best approach for this?

Thanks!

**Comment by KittyChiu** [comment]
Not absolutely disallow contributions, but there are branch protection rules you can try to protect your branches
- `Lock branch` if you want to make it read-only
- `Restrict who can push to matching branches` to assign only you can push

Alternatively, change your repo visibility to `private`.

**Reply by PabloPomodoro** [reply]
Thank you for your quick reply!

I definitely don't want to make my repo private, since I want to on the one hand, have people that use my app know, what is in it, and on the other hand (hopefully) set a good example for best pratctices and code quality :)

Please correct me if I'm wrong, but currently my repo should be set up in a way that only I can contribute directly, but I found that there is no functionality to disable the PR tab in repos, so people could submit PRs, which I would have to decline by hand.

The licence is important to me, because I want everybody to either build on my repo, or host it themselves or use the code in any way they would like to; except for contributing to my repo directly (it would obviously fine, if they forked it and built on their repo going forward)

Your thoughts on which licence would suit my desires best (if any) are much appreciated! :)

**Reply by KittyChiu** [reply]
If you want to be the sole contributor with no PR or branches created by others, making it `private` is the way to go. ~~Currently you cannot prevent people creating branches and raise PRs, except for custom cleanup automations.~~

UPDATED: You are right, my mistake on your workflows.

This [site](https://choosealicense.com/) gives some good guidance on which license to choose. Hope it helps.

**Reply by PabloPomodoro** [reply]
Thank you for your reply.

However I don't think you're correct on this - I just now tested with my other github user and I was not able to create PRs, branches, let alone modified or delete files directly.
Also I'm not going to set my repo to private since I want to share my work and enable other users to fork my repo.

But my question is about the fitting licence for my project - thank you for the link; however I've already looked that this, but couldn't figure out, which licence would fit my scenario best.

I think I'll just add a MIT licence, unless you would have point against it?

Your help is much appreciated!

**Comment by ebndev** [comment]
Hi @PabloPomodoro,

Thank you for being a part of the GitHub community!

Although I am unable to provide legal advice on what licenses to obtain, I wanted to note that an MIT license is one of the most unrestrictive open-source licenses, so that may not be the correct license to add since you are trying to restrict users ability to create PRs.

You can read more about the open-source licenses [here](https://semaphoreci.com/blog/open-source-licensing#:~:text=The%20MIT%20License%20is%20one,can%20do%20with%20the%20code.) and you might find this issue helpful: [License forbids modification, but pull requests accepted](https://github.com/numworks/epsilon/issues/38#top).

Best of luck!

**Reply by PabloPomodoro** [reply, marked as answer]
Hi @ebndev ,

Thanks for your answer! After reading the blog article you linked and skimming through the github issue (intresting read!), I re-read the MIT licence and found that it doesn't address contributions at all. IANAL but I don't see any way how this licence could force me to allow contributions to my code base.

So I think I'll add a MIT licence to my repo. 🙂
